docs/architecture/adr-008-caching-handling.md

```diff
@@ -0,0 +1,202 @@
+We do NOT make use of cache tags in the configuration fetching but still we make
+use of the "use cache" directive.
+According to the [Nextjs documentation](https://nextjs.org/docs/app/api-reference/functions/cacheLife#default-cache-profiles)
+the default cache lifetime should be around **15 minutes** before it is revalidated.
+
+#### 3rd party service data
+
+1. **FBI GraphQL API**: Search functionality and bibliographic data
+2. **FBS REST API**: Patron services, loans, reservations
+3. **Publizon/PubHub**: E-book services
+4. **Unilogin**: Authentication services
+
+Go fetches data from several 3rd party services. All the requests uses either
+react query QueryClient - either server side or client side. Both query clients
+makes use of the function `getQueryClientStaleTime()` which is, at the moment of
+this writing, is set to **1 minute**. Unless the queryclient is told to
+invalidate certain query keys the cache will live for that amount of time.
+
+## Usage
+
+### Implementation Examples
+
+#### Page Caching with Cache Tags
+
+```typescript
+// Category page example
+async function getPage(slug: string[]) {
+  "use cache"
+  const {
+    go: { cacheTags },
+    ...data
+  } = await loadPageData({
+    contentPath: slug.join("/"),
+    type: "category",
+  })
+
+  if (cacheTags) {
+    cacheTag(...cacheTags)
+  }
+
+  return { go: { cacheTags }, ...data }
+}
+```
+
+#### Configuration Caching
+
+```typescript
+const getDplCmsPrivateConfigData = async () => {
+  "use cache"
+  // Default cache time: ~15 minutes
+  try {
+    const data = await queryDplCmsPrivateConfig()
+    return privateConfigSchema.parse(data)
+  } catch {
+    // Fallback configuration
+    return defaultConfig
+  }
+}
+```
+
+#### React Query Setup
+
+```typescript
+// Server-side QueryClient
+const getQueryClient = cache(
+  () =>
+    new QueryClient({
+      defaultOptions: {
+        queries: {
+          staleTime: getQueryClientStaleTime(), // 1 minute in production, 0 in dev
+        },
+      },
+    })
+)
+
+// Client-side QueryClient
+const [client] = useState(
+  new QueryClient({
+    defaultOptions: {
+      queries: {
+        staleTime: getQueryClientStaleTime(),
+      },
+    },
+  })
+)
+```
+
+#### 3rd Party Service Queries
+
+```typescript
+// FBI Search API example
+export const performSearch = fromPromise(
+  ({ input: { q, filters, offset, limit, queryClient } }) => {
+    const args = { q: { all: q }, offset, limit, filters }
+    return queryClient.fetchQuery({
+      queryKey: useSearchWithPaginationQuery.getKey(args),
+      queryFn: useSearchWithPaginationQuery.fetcher(args),
+    })
+  }
+)
+```
+
+### Cache Invalidation
+
+#### Revalidation Endpoint
+
+FB CMS triggers cache invalidation by calling:
+
+```
+GET /api/cache/revalidate?secret={DRUPAL_REVALIDATE_SECRET}&tags=tag1,tag2,tag3
+```
+
+#### Manual Invalidation
+
+```typescript
+// In React Query
+queryClient.invalidateQueries({ queryKey: ["search", searchTerm] })
```

Comment:
Consider if a clearer relationship between the example here and in "3rd Party Service Queries" can be make.

Should this use something like ` useSearchWithPaginationQuery.getKey(searchTerm)`?